tiledb/sm/subarray/subarray.h

```diff
@@ -117,32 +118,65 @@ class DenseTileSubarray {
     }
   };
 
+  /** The type of the allocator. Required to make the type allocator-aware.
+   *
+   * uint8_t was arbitrarily chosen and does not matter; allocators can convert
+   * between any types.
+   */
+  using allocator_type = tdb::pmr::polymorphic_allocator<uint8_t>;
+
   /* ********************************* */
   /*     CONSTRUCTORS & DESTRUCTORS    */
   /* ********************************* */
 
   DenseTileSubarray() = delete;
 
-  DenseTileSubarray(unsigned dim_num)
-      : ranges_(dim_num) {
+  DenseTileSubarray(unsigned dim_num, const allocator_type& alloc = {})
+      : original_range_idx_(alloc)
+      , ranges_(dim_num, alloc) {
+  }
+
+  /** Default copy constructor. */
+  DenseTileSubarray(const DenseTileSubarray&) = default;
+  /** Default move constructor. */
+  DenseTileSubarray(DenseTileSubarray&&) = default;
+
+  /** Allocator-aware copy constructor. */
+  DenseTileSubarray(const DenseTileSubarray& other, const allocator_type& alloc)
+      : original_range_idx_(other.original_range_idx_, alloc)
+      , ranges_(other.ranges_, alloc) {
   }
 
+  /** Allocator-aware move constructor. */
+  DenseTileSubarray(DenseTileSubarray&& other, const allocator_type& alloc)
+      : original_range_idx_(std::move(other.original_range_idx_), alloc)
+      , ranges_(std::move(other.ranges_), alloc) {
+  }
+
```

Comment:
I remember being asked to remove these allocator-aware copy/move constructors from `Range` when I added tracking to it, but if I remove them from here compiling will fail.